--- tracker-0.10.5/configure.ac.wiget	2011-03-29 20:56:42.080059721 +0200
+++ tracker-0.10.5/configure.ac	2011-03-29 20:57:08.265734956 +0200
@@ -1132,7 +1132,7 @@ fi
 
 if test "x$have_tracker_miner_evolution" = "xyes"; then
 
-   PKG_CHECK_MODULES(MINER_EVOLUTION_SHELL, evolution-shell > 2.91,
+   PKG_CHECK_MODULES(MINER_EVOLUTION_SHELL, evolution-shell-3.0 > 2.91,
                      [miner_evolution_2_91=yes],
                      [miner_evolution_2_91=no])
 
