--- tracker-2.0.1/src/libtracker-data/Makefile.am.link	2017-11-05 17:23:29.529766471 +0100
+++ tracker-2.0.1/src/libtracker-data/Makefile.am	2017-11-05 17:23:01.347991992 +0100
@@ -57,6 +57,7 @@ libtracker_data_la_SOURCES =
 libtracker_data_la_LIBADD =                            \
 	$(top_builddir)/src/gvdb/libgvdb.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la \
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql-intermediate-vala.la \
 	$(BUILD_LIBS)                                  \
 	$(LIBTRACKER_DATA_LIBS)                        \
 	-lm
