--- tracker-1.10.4/docs/reference/libtracker-sparql/Makefile.am.orig	2017-01-28 20:17:29.984735155 +0100
+++ tracker-1.10.4/docs/reference/libtracker-sparql/Makefile.am	2017-01-28 20:17:32.331401795 +0100
@@ -56,6 +56,8 @@
         --extra-dir=$(GLIB_PREFIX)/share/gtk-doc/html/glib \
         --extra-dir=$(GLIB_PREFIX)/share/gtk-doc/html/gio
 
+export LD_LIBRARY_PATH=$(top_builddir)/src/libtracker-sparql-backend/.libs:$(top_builddir)/src/libtracker-data/.libs:$(top_builddir)/src/libtracker-common/.libs
+
 include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
