diff -urN tracker-0.6.6/src/tracker-preferences/tracker-preferences.c tracker-0.6.6.new/src/tracker-preferences/tracker-preferences.c
--- tracker-0.6.6/src/tracker-preferences/tracker-preferences.c	2008-02-03 04:36:57.000000000 +0100
+++ tracker-0.6.6.new/src/tracker-preferences/tracker-preferences.c	2008-04-17 19:31:09.000000000 +0200
@@ -995,7 +995,7 @@
 	if (!path)
 		return;
 
-	if (!strcasecmp (path, g_get_home_dir ())) {
+	if (!strcmp (path, g_get_home_dir ())) {
 		item = glade_xml_get_widget (priv->gxml,
 					     "chkIndexHomeDirectory");
 		gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (item), TRUE);
@@ -1091,7 +1091,7 @@
 			gtk_tree_model_get (GTK_TREE_MODEL (model), &iter, 0,
 					    &value, -1);
 
-			if (!strcasecmp (item, value))
+			if (!strcmp (item, value))
 				return;
 		} while (gtk_tree_model_iter_next
 			 (GTK_TREE_MODEL (model), &iter));
