--- tracker-1.0.2/src/tracker-extract/tracker-extract-gif.c.orig	2014-05-28 10:47:09.000000000 +0200
+++ tracker-1.0.2/src/tracker-extract/tracker-extract-gif.c	2014-07-11 20:26:35.476861511 +0200
@@ -684,12 +684,15 @@
 
 	g_free (uri);
 
-	if (DGifCloseFile (gifFile) != GIF_OK) {
 #if GIFLIB_MAJOR < 5
+	if (DGifCloseFile (gifFile) != GIF_OK)
 		print_gif_error ();
-#else  /* GIFLIB_MAJOR < 5 */
+#elif (GIFLIB_MAJOR == 5) && (GIFLIB_MINOR == 0)
+	if (DGifCloseFile (gifFile) != GIF_OK)
 		gif_error ("Could not close GIF file", gifFile->Error);
-#endif /* GIFLIB_MAJOR < 5 */
-	}
+#else  /* GIFLIB >= 5.1 */
+	if (DGifCloseFile (gifFile, &err) != GIF_OK)
+		gif_error ("Could not close GIF file", err);
+#endif /* GIFLIB_MAJOR ... */
 
 	return TRUE;
